actions/helm_chart_e2e.go

```diff
@@ -0,0 +1,130 @@
+package actions
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/google/go-github/github"
+)
+
+const (
+	// RepoFlag represents the '-repo' flag
+	RepoFlag = "repo"
+	// RefFlag represents the '-ref' flag (for specifying a SHA, branch or tag)
+	RefFlag = "ref"
+)
+
+// HelmChartE2E is the cli handler for generating a release helm chart for deis-e2e
+func HelmChartE2E(ghClient *github.Client) func(*cli.Context) {
+	return func(c *cli.Context) {
+		chartDir := "workflow-dev-e2e"
+		downloadFiles(ghClient, c, chartDir, []string{"README.md", "Chart.yaml"})
+
+		err := updateFilesWithRelease(chartDir)
+		if err != nil {
+			log.Fatalf("Error updating files with release (%s)", err)
+		}
+	}
+}
+
+func createSubDir(dirName string) {
+	dir := fmt.Sprintf("%s/%s", stagingDirName, dirName)
+	if _, err := os.Stat(dir); os.IsNotExist(err) {
+		err := os.Mkdir(dir, os.ModePerm)
+		if err != nil {
+			log.Fatalf("Error creating dir %s (%s)", dir, err)
+		}
+	}
+}
+
+func downloadFiles(ghClient *github.Client, c *cli.Context, rootDir string, fileNames []string) {
+	var opt github.RepositoryContentGetOptions
+	opt.Ref = c.GlobalString(RefFlag)
+
+	createSubDir(rootDir)
+
+	for _, fileName := range fileNames {
+		filePath := fmt.Sprintf("%s/%s", rootDir, fileName)
+		readCloser, err := downloadContents(ghClient, c.GlobalString(OrgFlag), c.GlobalString(RepoFlag), filePath, &opt)
+		if err != nil {
+			log.Fatalf("Error downloading contents of %s (%s)", filePath, err)
+		} else {
```

Comment:
no need to put everything here inside an `else` because the `log.Fatalf` will kill execution